katsdpimager/loader_katdal.py

```diff
@@ -55,6 +56,13 @@ def _timestamp_to_fits(timestamp):
     return Time(float(timestamp), format='unix').utc.isot
 
 
+def _compute(*arrays):
+    """Like da.compute, but work around https://github.com/dask/dask/issues/3595."""
```

Comment:
That dask issue was actually fixed quite a while ago (I mostly cargo-culted this from DaskLazyIndexer). My suspicion is that this wrapper will still improve performance (by computing directly into the target instead of building intermediate chunks then concatenating them), but since I don't know what the fix for that issue was I should still test it.

Reply:
I've given it a try - using this wrapper is almost twice as fast! I'll open a new Dask issue and modify the docstring.

Reply:
So is the main difference that this function does not concatenate chunks but leave them as a list of chunks?

Reply:
No, you still get ndarrays out either way. The difference is that with this approach, each chunk gets written into the final array as it is computed, where with `compute` it first computes all the chunks, then concatenates them.

Reply:
Your scheme seems more sensible :-)